pyeudiw/trust/__init__.py

```diff
@@ -27,15 +29,30 @@ def __init__(self, storage: DBEngine, httpc_params, trust_anchor: str = None, **
 
     @property
     def evaluation_method(self) -> bool:
-        # TODO: implement automatic detection of trust evaluation
-        # method based on internal trust evaluetion property
-        # TODO: implement the detect of x509 trust evaluation method here
-        return self.federation
+        try:
+            getattr(self, "typ")
+        except Exception:
+            raise MissingTrustType(
+                "Unknown Trust Type: can't find 'typ' in headers"
+            )
+
+        if self.typ.upper() == "JWT":
+            return self.federation
+        elif self.typ.upper() == "X509":
+            return self.x509
+        else:
+            raise InvalidTrustType(
+                "Unknown Trust Type: trust type not supported"
+                f"Found {self.typ}"
+            )
     
     def _retrieve_anchor(self):
-        _first_statement = unpad_jwt_payload(self.trust_chain[-1])
-        trust_anchor_eid = self.trust_anchor or _first_statement.get(
-            'iss', None)
+        if self.typ.upper() == "JWT":
+            _first_statement = unpad_jwt_payload(self.trust_chain[-1])
+            trust_anchor_eid = self.trust_anchor or _first_statement.get(
+                'iss', None)
+        else:
+            trust_anchor_eid = self.trust_anchor or get_issuer_from_x5c(self.x5c)
```

Comment:
self.trust_anchor should be valued with federation or x509 in the constructor, as it was for federation